### ISSUE TITLE
Kotlin properties with "is" and "get" prefixed are renamed during POJO serialization

### ISSUE THREAD

**Issue by francos** [firebase/firebase-admin-java#472]
### Describe your environment

  * Operating System version: MacOS 10.15.5
  * Firebase SDK version: 6.12.2
  * Library version: 6.12.2
  * Firebase Product: Firestore

### Describe the problem

Hi, I'm seeing inconsistent behaviour when saving and reading from Firestore using the Java Admin SDK (using Kotlin).
I'm saving a document that looks like this:

```kotlin
data class Whatever(
    val result: Result? = null
)
 
data class Result(
    val isSuccess: Boolean = false
)
```

When I save this to Firestore like this:

```kotlin
val whatever = Whatever(result = Result(isSuccess = true))

collection.document(docId).set(whatever).get()
```

then the `isSuccess` property is saved as a `success` field in Firestore (I verified this by checking the field on the Firebase Console).

However, when I want to read this same document from Firestore like this:

```kotlin
collection.document(docId).get().get().toObject(Whatever::class.java)
```

then the `isSuccess` property is not populated with `true` and the default `false` is returned.

Am I doing something wrong here or is there an inconsistency in how the Firebase SDK reads and writes data?

**Comment by schmidt-sebastian** [comment]
Thanks for filing this. I will try to repro this either today or tomorrow and can get back to you with an answer then.

**Comment by francos** [comment]
Thanks for getting back to me and for the explanation @schmidt-sebastian, I thought it was something like that as I've seen similar behaviours on other no-SQL databases :).

Just FYI, something weird happened today, as I was about to change the name of the variable from `isSuccess` to `success` I tried this again and suddenly it worked properly 🤔. Did you change anything on the server-side that fixed this by any chance? (I am now using version `6.16.0` of the admin SDK but I also tried this with that version last week and I still saw the error so it can't be a library fix).

**Comment by schmidt-sebastian** [comment]
The serialization is purely on the SDK side and the backend is unaware of these prefixes. I don't have an answer for this mystery.

**Comment by francos** [comment]
I assumed it was done on the SDK side which makes this very strange 😆. Thanks for the help @schmidt-sebastian. I will keep this value as `success` for now just in case.

It'd still be great if this could be fixed as I'm not sure if the web and iOS SDKs behave the same way or if they allow for the `is` at the beginning (in which case this could bring problems when writing and reading data from different apps on different platforms for people that are not aware of this).

**Comment by schmidt-sebastian** [comment]
The Web SDK doesn't have any object serialization built in. Instead, it defines a converter interface that the user has to implement. iOS uses Codeable for its conversion, which  doesn't special-case methods with prefixes.

**Comment by francos** [comment]
Thanks for the update on the other SDKs @schmidt-sebastian. 

In that case, seems like indeed this needs to be fixed to keep consistency between SDKs and avoid possible read issues when something is written from the Java SDK and then read from the web and/or iOS or vice-versa.

**Comment by schmidt-sebastian** [comment]
This behavior is a Java-specific behavior that aims to ease the use of POJOs in our API. Once the objects are translated to simple Firestore maps, the behavior is consistent.

**Comment by hiranya911** [comment]
I believe this issue has been sufficiently handled. Please report any follow ups directly at https://github.com/googleapis/nodejs-firestore

**Comment by francos** [comment]
Hi @hiranya911 and @schmidt-sebastian, I feel like this has not been sufficiently handled.

I tested the following scenario today and it doesn't seem to work:

1. Create a document from the JS SDK using an object with a property that starts with `is` (e.g., `{ "isSuccess": true }`).

2. Read the document from the Java SDK and convert it to an object using the `toObject()` method.

When I try to read the object on step 2 the app fails to convert the document to a Java object because the JS SDK saved the name of the field as `isSuccess` and there is no way for the Java SDK to convert that apparently. I tried naming the property in the Kotlin class both `isSuccess` and `success` and they both failed to get the proper value from the underlying document field.

I understand that fixing this would probably break some backward compatibility things but it seems like it has to be fixed at some point to keep consistency between the SDKs and also so that values saved from one SDK and read by the other work properly in all possible scenarios. 

Also, not sure why you added a link to the NodeJS lib @hiranya911 as this seems to be an inconsistency with the Java SDK.
